mocca-client/src/main/java/com/paypal/mocca/client/MoccaSerializer.java

```diff
@@ -369,13 +377,57 @@ private void writeSelectionSet(final ByteArrayOutputStream requestPayload, final
         }
     }
 
+    /**
+     *
+     * Writes the selection set of the GraphQL request message using the user provided SelectionSet annotation with Ignore attribute set as reference.
+     *
+     * It uses the Response type to start an initial list which is then filtered if the value is present in the ignore list
+     *
+     * @param requestPayload the output stream object used to write the request payload, based on the other parameters
+     * @param ignoreFields The list of fields which have to be ignored in selection set generation
+     * @param responseType the return type set in the GraphQL operation method, useful when defining the request selection set
+     */
+    private void writeUserProvidedSelectionSetUsingIgnore(ByteArrayOutputStream requestPayload,
```

Comment:
I was expecting you to filter fields inside of method `writeResponseTypeSelectionSet` (similarly to how we filter ignore fields set in the GraohQL variables), instead of adding a brand new method just for this use case.

This method you added here might be functional for simple scenarios, but it cerainly lacks support for many corner cases (depending on parameterized types) that are already addressed in `writeResponseTypeSelectionSet`. Besides that, we are duplicating code here.

Having said that, please remove this method, and add the ignore logic inside `writeResponseTypeSelectionSet`. You will have to add a new parameter to it passing the fields to be ignored, similarly to what was done in the GraphQL variables ignore mechanism. I hope this makes sense, let me know if not.

Reply:
Changed the code accordingly
Only issue I think in this is I have to pass an empty list in cases when value and ignore both are absent and Response type is used.
Please review and we can discuss on if there is even a clean way than this :)